MFVNeutralino/plugins/Vertexer.cc

```diff
@@ -1,891 +1,1778 @@
+  bool shared_jet = std::find_first_of(sv_match_tracktojet_which_jetidx[vtx0idx].begin(), sv_match_tracktojet_which_jetidx[vtx0idx].end(), sv_match_tracktojet_which_jetidx[vtx1idx].begin(), sv_match_tracktojet_which_jetidx[vtx1idx].end()) != sv_match_tracktojet_which_jetidx[vtx0idx].end();
+
+  std::vector<std::vector<size_t>> sv_lonesharedtrack_which_trkidx;
+  std::vector<size_t> sv0_lonesharedtrack_which_trkidx;
+  std::vector<size_t> sv1_lonesharedtrack_which_trkidx;
+
+  if (shared_jet) {
+
+    int nsharedjets = 0;
+
+    std::vector<std::vector<int>> sv_match_tracktojet_which_temp_jetidx = sv_match_tracktojet_which_jetidx;
+    std::vector<std::vector<int> >sv0_match_sharedjettotrack_which_trkidx;
+    std::vector<std::vector<int> >sv1_match_sharedjettotrack_which_trkidx;
+
+    std::vector<int> sv0_match_tracktojet_which_jetidx = sv_match_tracktojet_which_jetidx[vtx0idx];
+    std::vector<int> sv0_match_tracktojet_which_trkidx = sv_match_tracktojet_which_trkidx[vtx0idx];
+    std::vector<int> sv0_match_sharedjettotrack_which_temp_trkidx;
+
+    std::vector<int> sv1_match_tracktojet_which_jetidx = sv_match_tracktojet_which_jetidx[vtx1idx];
+    std::vector<int> sv1_match_tracktojet_which_trkidx = sv_match_tracktojet_which_trkidx[vtx1idx];
+    std::vector<int> sv1_match_sharedjettotrack_which_temp_trkidx;
+
+    while (std::find_first_of(sv_match_tracktojet_which_temp_jetidx[vtx0idx].begin(), sv_match_tracktojet_which_temp_jetidx[vtx0idx].end(), sv_match_tracktojet_which_temp_jetidx[vtx1idx].begin(), sv_match_tracktojet_which_temp_jetidx[vtx1idx].end()) != sv_match_tracktojet_which_temp_jetidx[vtx0idx].end()) {
+      nsharedjets++;
+      std::vector<int>::iterator it = std::find_first_of(sv_match_tracktojet_which_temp_jetidx[vtx0idx].begin(), sv_match_tracktojet_which_temp_jetidx[vtx0idx].end(), sv_match_tracktojet_which_temp_jetidx[vtx1idx].begin(), sv_match_tracktojet_which_temp_jetidx[vtx1idx].end());
+      int it_idx = std::distance(sv_match_tracktojet_which_temp_jetidx[vtx0idx].begin(), it);
+      int jet_index = sv_match_tracktojet_which_temp_jetidx[vtx0idx].at(it_idx);
+
+      sv_match_tracktojet_which_temp_jetidx[vtx0idx].erase(std::remove(sv_match_tracktojet_which_temp_jetidx[vtx0idx].begin(), sv_match_tracktojet_which_temp_jetidx[vtx0idx].end(), jet_index), sv_match_tracktojet_which_temp_jetidx[vtx0idx].end());
+      sv_match_tracktojet_which_temp_jetidx[vtx1idx].erase(std::remove(sv_match_tracktojet_which_temp_jetidx[vtx1idx].begin(), sv_match_tracktojet_which_temp_jetidx[vtx1idx].end(), jet_index), sv_match_tracktojet_which_temp_jetidx[vtx1idx].end());
+
+      // start collecting shared tracks of sv0 for each shared jet
+      std::multimap<int, size_t> sv0_m;
+      for (size_t k = 0; k < sv0_match_tracktojet_which_jetidx.size(); k++)
+        if (sv0_match_tracktojet_which_jetidx[k] == jet_index) { sv0_m.insert({ sv0_match_tracktojet_which_jetidx[k], k }); }
+
+      for (auto it = sv0_m.begin(); it != sv0_m.end(); )

```

Comment:
We reused the variable name `it`, we should be careful here and use different names.

Reply:
So, I will change  `it` to `sharedjet_it` and  `it_idx` to `sharedjet_it_idx.`